connection.go

```diff
@@ -1567,20 +1472,20 @@ func (conn *Connection) newWatcherImpl(key string, callback WatchCallback) (Watc
 	}, nil
 }
 
-// ServerProtocolVersion returns protocol version and protocol features
+// ServerProtocolInfo returns protocol version and protocol features
 // supported by connected Tarantool server. Beware that values might be
 // outdated if connection is in a disconnected state.
 // Since 1.10.0
 func (conn *Connection) ServerProtocolInfo() ProtocolInfo {
 	return conn.serverProtocolInfo.Clone()
 }
 
-// ClientProtocolVersion returns protocol version and protocol features
+// ClientProtocolInfo returns protocol version and protocol features
 // supported by Go connection client.
 // Since 1.10.0
 func (conn *Connection) ClientProtocolInfo() ProtocolInfo {
 	info := clientProtocolInfo.Clone()
-	info.Auth = conn.opts.Auth
+	info.Auth = conn.serverProtocolInfo.Auth
```

Comment:
I think we could remove the method. The information could be saved by the client before the connection creation.